service/src/test/java/com/ge/predix/controller/test/ResourcePrivilegeMgmtWithGraphDbControllerIT.java

```diff
@@ -0,0 +1,150 @@
+/*******************************************************************************
+ * Copyright 2016 General Electric Company.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package com.ge.predix.controller.test;
+
+import static com.ge.predix.acs.testutils.Constants.RESOURCE_EVIDENCE_0_ATTRS_0;
+import static com.ge.predix.acs.testutils.Constants.RESOURCE_EVIDENCE_0_ID;
+import static com.ge.predix.acs.testutils.Constants.RESOURCE_SITE_0_ATTRS_0;
+import static com.ge.predix.acs.testutils.Constants.RESOURCE_SITE_0_ID;
+import static com.ge.predix.acs.testutils.Constants.RESOURCE_XFILE_0_ATTRS_0;
+import static com.ge.predix.acs.testutils.Constants.RESOURCE_XFILE_0_ID;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.web.context.WebApplicationContext;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.ge.predix.acs.rest.BaseResource;
+import com.ge.predix.acs.rest.Parent;
+import com.ge.predix.acs.rest.Zone;
+import com.ge.predix.acs.testutils.MockAcsRequestContext;
+import com.ge.predix.acs.testutils.MockMvcContext;
+import com.ge.predix.acs.testutils.MockSecurityContext;
+import com.ge.predix.acs.testutils.TestUtils;
+import com.ge.predix.acs.utils.JsonUtils;
+import com.ge.predix.acs.zone.management.ZoneService;
+
+@WebAppConfiguration
+@ContextConfiguration("classpath:controller-tests-context.xml")
+@ActiveProfiles(profiles = { "h2", "public", "simple-cache", "titan" })
+public class ResourcePrivilegeMgmtWithGraphDbControllerIT extends AbstractTestNGSpringContextTests {
```

Comment:
This does not need to be an integration test. If there are additional testcases from GraphResourceRepositoryTest, can we add them as unit tests ?